docs/source/howto/installation.rst

```diff
@@ -18,7 +18,149 @@ Managing profiles
 Configuring your installation
 =============================
 
-`#4001`_
+.. _how-to:installation:configure:instance-isolation:
+
+Isolating multiple instances
+----------------------------
+An AiiDA instance is defined as the installed source code plus the configuration folder that stores the configuration files with all the configured profiles.
+It is possible to run multiple AiiDA instances on a single machine, simply by isolating the code and configuration in a virtual environment.
+This section will explain how to accomplish this.
+
+To isolate the code, simply create a virtual environment with the tool of your choosing :ref:`SHOULD THERE BE A LINK HERE?<>` and :ref:`run the installation<intro/install_advanced>` while it is activated.
+Whenever you activate this particular environment, you will be running the particular version of AiiDA (and all the plugins) that you installed specifically for it.
+
+Then there is the configuration of AiiDA, which is stored in the configuration directory which is always named ``.aiida`` and by default is stored in the home directory.
+Therefore, the default path of the configuration directory is ``~/.aiida``.
+When unchanged, each AiiDA instance (each installation in a virtual environment) will store the profiles that are configured for it in this same folder.
+However, often one would like to keep the profile for each instance separate.
```

Comment:
Maybe, we should just use stronger language here? It should be clear that separating the ``.aiida`` folders is the "default" thing to do, and you need to be careful if you don't. 

For my understanding, if the ``.aiida`` folders are _not_ split, the user needs to make sure to only ever use the "right" profiles with a given virtualenv, right?

Reply:
I'm ok with using stronger language or leaving as is. It seems to me like the possibility of a version mismatch should actually be addressed programatically.

Reply:
Hmm, not sure if there is a good way to catch this programmatically. The problems I can see happening are:
- Using two different AiiDA versions: The newer AiiDA version can trigger a DB migration, after which the older version will complain. The question here is how to distinguish it from a "normal" upgrade.
- Different plugin / dependency versions: This is problematic when the daemon and the launching process run on different versions. The persisted process can probably  be "wrong" in countless ways.

One way I could see of handling this would be pinning either ``which verdi`` or ``which python`` in the profile - if it changes, flash a big warning and ask for confirmation to change it.

Nevertheless, I think it should be made clear in the documentation - ideally the programmatic catch is there only as a safety net, and users are led to do the right thing.

Reply:
I tend to agree with @greschd that we should guide the user to do the right thing and just have a safety net. We already have a check for `aiida-core` that will check the version of the database schema against that of the code to check that they are compatible. However, this does not (and cannot) check the code of the plugins. Moreover, there is currently a way that the check can be circumvented (by accident). If you install a new version of AiiDA and configure a profile for an existing database using `verdi setup` the command will automatically migrate it. Since new databases always have to be migrated, this is done automatically and unprompted. This can be unintended and therefore unexpected, especially given the reverse migration is not possible. Now this is of course a bug of sorts that we should perhaps fix (only automatically migrate with `verdi setup` if the database is "empty"), but my point is that there might be more subtle problems like this that we might not have thought off. TLDR: I think it is indeed wise to make the distinction that Dominik described, just for completeness and clarity.

Reply:
I think this point is resolved with the changes made, I'll let @csadorf review the rest.